core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -758,20 +758,6 @@ protected Map<String, String> summary() {
     return summaryBuilder.build();
   }
 
-  /**
-   * Apply the update's changes to the base table metadata and return the new manifest list.
-   *
-   * @param base the base table metadata to apply changes to
-   * @return a manifest list for the new snapshot.
-   * @deprecated Will be removed in 1.2.0, use {@link MergingSnapshotProducer#apply(TableMetadata,
-   *     Snapshot)}.
-   */
-  @Deprecated
-  @Override
-  public List<ManifestFile> apply(TableMetadata base) {
```

Comment:
Why remove these? It doesn't seem like a big deal to keep the versions that default the parent snapshot to current.

Reply:
This is code that changed this method in #4926. We then restored+deprecated the old API in #6146 for one minor release and are removing it here now. 
Let me know if we'd want to keep those methods, then we can just drop the `Deprecated` annotations from those